db/init.sql

```diff
@@ -4,13 +4,315 @@ CREATE DATABASE mm_db;
 CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
 
 CREATE TYPE user_role AS ENUM ('admin', 'teacher', 'student');
+CREATE TYPE attempt_state AS ENUM ('in_progress', 'completed');
+-- NOTE(nrydanov): Need to think of other types together
+CREATE TYPE block_type AS ENUM ('assignment', 'text', 'media', 'quiz');
+
 
 CREATE TABLE users (
     id uuid NOT NULL PRIMARY KEY,
+    -- Auth data
     email varchar(128) UNIQUE NOT NULL,
+    password bytea NOT NULL,
+    -- Identity data
     name varchar(32) NOT NULL,
     surname varchar(32) NOT NULL,
     patronymic varchar(32),
-    role user_role NOT NULL,
-    password bytea NOT NULL
+    date_of_birth date,
+    
+    -- University data
+    -- NOTE(nrydanov): This data should be enough to determine exact academic group
+    faculty_id uuid NOT NULL,
+    specialty_id uuid NOT NULL,
+    admission_date date NOT NULL,
+    -- Service info
+    -- NOTE(nrydanov): Can be used to make admin a teacher for a while
+    role_override user_role,
+    created_at timestamp NOT NULL
+);
+
+
+CREATE TABLE course (
+    id uuid NOT NULL PRIMARY KEY,
+    creator_id uuid NOT NULL,
+    name varchar(64) NOT NULL,
+    -- Service info 
+    created_at timestamp NOT NULL
+);
+
+
+-- NOTE(nrydanov): May be helpful to filter out courses for certain faculty
+CREATE TABLE course_faculty (
+    course_id uuid NOT NULL,
+    faculty_id uuid NOT NULL
+);
+
+
+-- NOTE(nrydanov): May be helpful to auto-assign user for one of existing
+-- group within the course
+CREATE TABLE group_password (
+    password bytea NOT NULL,
+    group_id uuid NOT NULL PRIMARY KEY
+);
+
+
+CREATE TABLE user_group (
+    id uuid NOT NULL PRIMARY KEY,
+    name varchar(64) NOT NULL
+);
+
+CREATE TABLE course_user_group (
+    user_id uuid NOT NULL,
+    course_id uuid NOT NULL,
+    group_id uuid NOT NULL
+);
+
+
+CREATE TABLE block (
+    id uuid NOT NULL PRIMARY KEY,
+    type block_type NOT NULL,
+    -- NOTE(nrydanov): May be useful to create chain of blocks
+    next_block uuid,
+    -- NOTE(nrydanov): Contains different JSON data for each type of block
+    data jsonb NOT NULL
+);
+
+-- NOTE(nrydanov): group there isn't an academic one, but just multiple tasks
+-- on one theme. Maybe there could be a better naming.
+CREATE TABLE task_group (
```

Comment:
```suggestion
CREATE TABLE task_topic (
```

As I said, `topic` could be a better naming in this case

Reply:
Yeah, let's rename it this way